packages/frontend/src/components/icon_button.tsx

```diff
@@ -1,19 +1,31 @@
-import { type JSX, splitProps } from "solid-js";
+import Tooltip from "@corvu/tooltip";
+import {type JSX, children, splitProps} from "solid-js";
 
 import "./icon_button.css";
 
 /** Styled, unobstrusive button intended to include an icon.
  */
 export function IconButton(
-    allProps: {
-        children: JSX.Element;
-    } & JSX.ButtonHTMLAttributes<HTMLButtonElement>,
+  allProps: {
+    children: JSX.Element;
+    tooltip?: string;
+  } & JSX.ButtonHTMLAttributes<HTMLButtonElement>
 ) {
-    const [props, buttonProps] = splitProps(allProps, ["children"]);
+  const [props, buttonProps] = splitProps(allProps, ["children", "tooltip"]);
 
-    return (
-        <button class="icon-button" {...buttonProps}>
-            {props.children}
-        </button>
-    );
+  return (
+    <Tooltip openOnHover>
+      <Tooltip.Trigger class="icon-button" {...buttonProps}>
+        {props.children}
+      </Tooltip.Trigger>
+      <Tooltip.Portal>
+        {props.tooltip && (
+          <Tooltip.Content data-corvu-tooltip-content={props}>
+            {props.tooltip}
+            <Tooltip.Arrow />
```

Comment:
We should remove the `Tooltip.Arrow` (i.e., delete this line) since the arrow is rendering on top of the button.